picard/ui/filebrowser.py

```diff
@@ -23,7 +23,51 @@
 from PyQt4 import QtCore, QtGui
 from picard import config
 from picard.formats import supported_formats
-from picard.util import find_existing_path
+from picard.util import find_existing_path, icontheme
+
+
+class FileBrowserPane(QtGui.QWidget):
+
+    def __init__(self, parent):
+        QtGui.QWidget.__init__(self, parent)
+        layout = QtGui.QVBoxLayout()
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(0)
+        self.toolbar = FileBrowserToolbar(self)
+        self.file_browser = FileBrowser(self)
+        layout.insertWidget(0, self.toolbar)
+        layout.insertWidget(1, self.file_browser)
+        self.setLayout(layout)
+        self.tagger = self.parent().tagger
+        self.toolbar.add_items_action.triggered.connect(self.add_items)
+
+    def save_state(self):
+        self.file_browser.save_state()
+
+    def restore_state(self):
+        self.file_browser.restore_state()
+
+    def _selected_paths(self):
+        paths = []
+        for index in self.file_browser.selectedIndexes():
+            paths.append(self.file_browser.model.filePath(index))
+        return paths
+
+    def add_items(self):
+        paths = self._selected_paths()
+        files = []
+        directories = []
+        for path in paths:
+            if os.path.isfile(path):
+                files.append(path)
+            else:
+                directories.append(path)
```

Comment:
Why not add directories directly here ?

I understand why you are building a list of files to pass to `tagger.add_files()`, but you may skip this for directories and just do `self.tagger.add_directory(path)` here.